### PR TITLE
S3C-654: check validity level vs. dump level

### PR DESCRIPTION
**Pull request by lamphamsy**
Check validity of log level vs. log dump level

### CODE REVIEW

File: lib/Config.js [hunk 1]
```diff
@@ -71,11 +71,24 @@ class Config {
...
+        // for check log level vs. log dump level
+        const newLogLevel = checkedConfig.level || this.logLevel;
+        const newLogDumpLevel = checkedConfig.dump || this.dumpThreshold;
+        if (newLogDumpLevel &&
+            !LogLevel.shouldLog(newLogDumpLevel, newLogLevel)) {
+            throw new Error(
+                'Logging level should be at most logging dump level');
+        }
+
+        if (checkedConfig.hasOwnProperty('level')) {
```
Comment (rahulreddy) [inline lib/Config.js]: This check is redundant as it's already checked at line 72.

File: tests/unit/Logger.js [hunk 2]
```diff
@@ -113,7 +113,7 @@ describe('WereLogs Logger is usable:', () => {
         const logger = new Logger('test');
         assert.doesNotThrow(
             () => {
-                logger.setLevel('fatal');
+                logger.setLevel('error');
```
Comment (rahulreddy) [inline tests/unit/Logger.js]: I think it makes more sense to add this is as another test. This way we are testing if it's working as expected when logLevel === dumpLevel and if logLevel < dumpLevel.

File: lib/Config.js (hunk 1 above)
Comment (lamphamsy) [inline lib/Config.js]: The previous check at line 72 is to check correct level value. This check here (at line 87) is to set this level value if it's given in the input config.

File: tests/unit/Logger.js (hunk 2 above)
Comment (lamphamsy) [inline tests/unit/Logger.js]: In fact, it's a quick fix as current log dump level is error, hence setLevel('fatal') throws an error :)

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by ThibaultRiviere** [comment]
@ironman-machine try

**Comment by ThibaultRiviere** [comment]
@ironman-machine r+

**Pull request merged**